bindings/swc_cli/src/commands/plugin.rs

```diff
@@ -231,55 +238,7 @@ build-wasm32 = "build --target wasm32-unknown-unknown"
         create_dir_all(&src_path)?;
         fs::write(
             &src_path.join("lib.rs"),
-            r##"use swc_core::{
-    ast::Program,
-    plugin::{plugin_transform, proxies::TransformPluginProgramMetadata},
-    testing_transform::test,
-    visit::{as_folder, FoldWith, VisitMut},
-};
-
-pub struct TransformVisitor;
-
-impl VisitMut for TransformVisitor {
-    // Implement necessary visit_mut_* methods for actual custom transform.
-    // A comprehensive list of possible visitor methods can be found here:
-    // https://rustdoc.swc.rs/swc_ecma_visit/trait.VisitMut.html
-}
-
-/// An example plugin function with macro support.
-/// `plugin_transform` macro interop pointers into deserialized structs, as well
-/// as returning ptr back to host.
-///
-/// It is possible to opt out from macro by writing transform fn manually
-/// if plugin need to handle low-level ptr directly via
-/// `__transform_plugin_process_impl(
-///     ast_ptr: *const u8, ast_ptr_len: i32,
-///     unresolved_mark: u32, should_enable_comments_proxy: i32) ->
-///     i32 /*  0 for success, fail otherwise.
-///             Note this is only for internal pointer interop result,
-///             not actual transform result */`
-///
-/// This requires manual handling of serialization / deserialization from ptrs.
-/// Refer swc_plugin_macro to see how does it work internally.
-#[plugin_transform]
-pub fn process_transform(program: Program, _metadata: TransformPluginProgramMetadata) -> Program {
-    program.fold_with(&mut as_folder(TransformVisitor))
-}
-
-// An example to test plugin transform.
-// Recommended streategy to test plugin's transform is verify
-// the Visitor's behavior, instead of trying to run `process_transform` with mocks
-// unless explicitly required to do so.
-test!(
-    Default::default(),
-    |_| as_folder(TransformVisitor),
-    boo,
-    // Input codes
-    r#"console.log("transform");"#,
-    // Output codes after transformed with plugin
-    r#"console.log("transform");"#
-);"##
-                .as_bytes(),
+            include_bytes!("../../../../crates/swc_plugin_testing/src/template.rs"),
```

Comment:
This will not work if this crate is published

Reply:
You are right. I forgot that crate can be installed and used separately.